src/query-builder/QueryExpressionMap.ts

```diff
@@ -404,12 +404,12 @@ export class QueryExpressionMap {
         map.cacheDuration = this.cacheDuration;
         map.relationPropertyPath = this.relationPropertyPath;
         map.of = this.of;
-        map.insertColumns = this.insertColumns;
-        map.whereEntities = this.whereEntities;
+        map.insertColumns = this.insertColumns.map((column) => column);
+        map.whereEntities = this.whereEntities.map((entity) => ({ ...entity }));
```

Comment:
I don't think we should change those. I think your particular problem is in `nativeParameters`, isn't it?

Reply:
okay, I've removed those changes